### ISSUE TITLE
libclc warning  operator '?:' has lower precedence than '|

### ISSUE THREAD

**Issue by LW-archlinux** [llvm/llvm-project#59209]
When building libclc against trunk rev b400dde4739f using GCC 12.2 the following warning is shown many times 

`warning: operator '?:' has lower precedence than '|'; '|' will be evaluated first [-Wbitwise-conditional-parentheses]`

Example:
```console
/home/panoramix/Documents/Aur/pkgbuilds/libclc-minimal-git/src/llvm-project/libclc/generic/lib/math/acosh.cl:120:48: warning: operator '?:' has lower precedence than '|'; '|' will be evaluated first [-Wbitwise-conditional-parentheses]
    ret = (ux & SIGNBIT_DP64) != 0UL | x < 1.0 ? as_double(QNANBITPATT_DP64) : ret;
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
/home/panoramix/Documents/Aur/pkgbuilds/libclc-minimal-git/src/llvm-project/libclc/generic/lib/math/acosh.cl:120:48: note: place parentheses around the '|' expression to silence this warning
    ret = (ux & SIGNBIT_DP64) != 0UL | x < 1.0 ? as_double(QNANBITPATT_DP64) : ret;
                                               ^
          (                                   )
/home/panoramix/Documents/Aur/pkgbuilds/libclc-minimal-git/src/llvm-project/libclc/generic/lib/math/acosh.cl:120:48: note: place parentheses around the '?:' expression to evaluate it first
    ret = (ux & SIGNBIT_DP64) != 0UL | x < 1.0 ? as_double(QNANBITPATT_DP64) : ret;
                                               ^
                                       (                                          )
1 warning generated.
```

Full at log https://app.box.com/s/gpgzk1s3kh70phdf4rlmzgmcq6z5lq5p
